periodic/periodic.go

```diff
@@ -174,39 +174,42 @@ func (r *RunnerOptions) Normalize() {
 		go func() {
 			gAbortMutex.Lock()
 			gOutstandingRuns++
+			n := gOutstandingRuns
 			runnerChan := r.Stop // need a copy to not race with assignement to nil
-			abortChan := gAbortChan
 			if gAbortChan == nil {
-				log.Infof("First outstanding run starting, catching signal")
+				log.Infof("WATCHER %d First outstanding run starting, catching signal", n)
 				gAbortChan = make(chan os.Signal, 1)
 				signal.Notify(gAbortChan, os.Interrupt)
 			}
+			abortChan := gAbortChan
```

Comment:
this is the serious bug fix, the rest of the changes are just to make logging easier to debug with a watcherId